libs/libcommon/src/libcommon/rows_utils.py

```diff
@@ -19,14 +19,15 @@ def transform_rows(
     cached_assets_base_url: str,
     cached_assets_directory: StrPath,
     offset: int,
+    row_idx_column: Optional[str],
 ) -> List[Row]:
     return [
         {
             featureName: get_cell_value(
                 dataset=dataset,
                 config=config,
                 split=split,
-                row_idx=offset + row_idx,
+                row_idx=offset + row_idx if row_idx_column is None else row[row_idx_column],
```

Comment:
`row[row_idx_column]` will be used for `/search` (because it cant depend on offset to get the id column, but it depends on `__hf_index_id`.
`offset + row_idx` is used by `/rows` (since it uses offset and will always have the same order for all rows)

Reply:
OK, perfect